cmd/otel-agent/subcommands/run/command.go

```diff
@@ -163,6 +162,12 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
 		fx.Invoke(func(_ collectordef.Component, _ defaultforwarder.Forwarder, _ optional.Option[logsagentpipeline.Component]) {
 		}),
 
+		configsyncimpl.OptionalModuleWithParams(),
```

Comment:
I understand that you added the `OptionalModuleWithParams` so you could update the `OptionalModule` to provide the params, and have `OptionalModuleWithParams` so we can pass the params ourselves. 

I'm wondering if we can achieve the same thing with just one function `OptionalModule`. @ogaca-dd currently is changing the way we pass params to the components. Example PR https://github.com/DataDog/datadog-agent/pull/28386

I think the configsync is a perfect candidate for that pattern

Reply:
Indeed this is a good candidate for this pattern.

Reply:
Agreed. As discussed will adress this in follow-up. Will add a `TODO` section and tackle by EOW.